src/components/Calculator.js

```diff
@@ -1,186 +1,86 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Calculator.css';
 import calculate from '../logic/calculator';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-    this.handleClick = this.handleClick.bind(this);
-  }
+function Calculator() {
+  const [state, setState] = useState({});
 
-  handleClick = (buttonName) => {
-    this.setState((state) => calculate(state, buttonName));
+  const handleClick = (e) => {
+    setState({ ...state, ...calculate(state, e.target.name) });
   };
```

Comment:
- [ ] Please avoid using the `function` keyword I recommend using `arrow function` while refactoring

Reply:
This issue is resolved! :+1: